### DISCUSSION TITLE
Support for SVC(Singing Voice Conversion) model?

### DISCUSSION THREAD

**Discussion by DLSeed** [civitai/civitai#286] [Ideas]
Currently SVC (Singing Voice Conversion) models are also hot, have you ever thought of supporting them? For example, support users to upload soVITS or DiffSvc singing voice conversion pretrained models, and the preview page can upload not only pictures, but also some song previews, which is convenient for users to choose models

**Comment by JustMaier** [comment]
I'd like the platform to be for all types of AI generative content, so it'd be great to add support for these models.

I haven't personally had a chance to play with these yet, so I have a few questions:
- What kind of files need to be uploaded for these?
- You mentioned pictures, but since this is primarily about generative audio samples, should the primary focus of the model pages be audio based?
- For browsing these, I think it make the most sense to have a small preview image with a play button the user can press to play a sample. Does that sound correct?
- How big are the models?
- What kind of hardware do you need to run the models?

Thanks for the suggestion, excited to see the platform expand beyond images

**Reply by DLSeed** [reply]
1.The SVC model changes the singer vocals of the source song into the target singer vocals you like. There are  known two types of  models, based on GAN and diffusion. soVITS (https://github.com/innnky/so-vits-svc/blob/32k/Eng_docs.md) is based on GAN, the model uploaded can be generator + discriminator - so that user can download it for inference  and further training, or just a simplified generator, then only can be used for song inference. DiffSvc (https://github.com/prophesier/diff-svc) is based on a diffusion model, similar to SD, with a main diffusion model and some auxiliary models, such as a Vocoder that turns Mel into a waveform.

2.Yes,model pages should be audio based, generally singers on the web have a more distinct 2D/3D avatar, adding her/his avatar image can better allow users to know whose voice the model is converting. And yes, a small preview image to indicate the identity of the singer is fine.

3.The size of the model depends on the setting of the model parameters. If the parameters the github link above are not changed, for GAN-based soVITS, D+G≈1GB, simplified G≈200MB; for diffusion model-based diffsvc, diffusion model+hubert+hifigan≈800MB

4.Using these models for inference is very fast, with a typical GPU (T4) on colab, it takes about 1 minute to infer a complete song. Training these models takes much longer, with the same colab GPU settings, it might take 20 hours or more.
